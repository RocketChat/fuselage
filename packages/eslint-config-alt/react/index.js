module.exports = {
  extends: '../minimal',
  plugins: ['react', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'jsx-quotes': ['error', 'prefer-single'],
    'react/display-name': 'error',
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-fragments': ['error', 'syntax'],
    'react/no-multi-comp': 'error',
    'react/react-in-jsx-scope': 'error',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
    'react': {
      version: 'detect',
    },
  },
  env: {
    browser: true,
    es6: true,
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.tsx'],
      extends: '../typescript',
      parser: '@typescript-eslint/parser',
      plugins: ['react', 'react-hooks'],
      rules: {
        'jsx-quotes': ['error', 'prefer-single'],
        'react/display-name': 'error',
        'react/jsx-uses-react': 'error',
        'react/jsx-uses-vars': 'error',
        'react/jsx-no-undef': 'error',
        'react/jsx-fragments': ['error', 'syntax'],
        'react/no-multi-comp': 'error',
        'react/react-in-jsx-scope': 'error',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
      },
      env: {
        browser: true,
        es6: true,
      },
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
        'react': {
          version: 'detect',
        },
      },
    },
    {
      files: [
        '**/*.stories.js',
        '**/*.stories.jsx',
        '**/*.stories.ts',
        '**/*.stories.tsx',
      ],
      rules: {
        'react/display-name': 'off',
        'react/no-multi-comp': 'off',
      },
    },
  ],
};
