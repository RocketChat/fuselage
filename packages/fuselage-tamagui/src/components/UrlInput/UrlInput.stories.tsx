import type { Meta, StoryObj } from '@storybook/react';
import { YStack, XStack, Text } from 'tamagui';

import { UrlInput } from './UrlInput';
import { Icon } from '../Icon';

const meta = {
  title: 'INPUTS/UrlInput',
  component: UrlInput,
  parameters: {
    layout: 'centered',
  },
  argTypes: {
    small: {
      control: 'boolean',
    },
    disabled: {
      control: 'boolean',
    },
    error: {
      control: 'text',
    },
    placeholder: {
      control: 'text',
    },
  },
  tags: ['autodocs'],
} satisfies Meta<typeof UrlInput>;

export default meta;
type Story = StoryObj<typeof UrlInput>;

export const Default: Story = {
  args: {
    'aria-label': 'URL',
  },
};

export const WithIconAddon: Story = {
  args: {
    'aria-label': 'URL with icon',
    addon: <Icon name='discover' size={20} />,
  },
};

export const Invalid: Story = {
  args: {
    'aria-label': 'URL with error',
    error: 'Error message',
  },
};

export const Disabled: Story = {
  args: {
    'aria-label': 'Disabled URL',
    disabled: true,
  },
};

export const WithPlaceholder: Story = {
  args: {
    'aria-label': 'URL with placeholder',
    placeholder: 'Enter URL...',
  },
};

export const WithValue: Story = {
  args: {
    'aria-label': 'URL with value',
    defaultValue: 'https://rocket.chat',
  },
};

export const Small: Story = {
  args: {
    'aria-label': 'Small URL',
    small: true,
    placeholder: 'Small URL...',
  },
};

export const States: Story = {
  render: () => (
    <YStack space={10} alignItems="center">
      <XStack space={10} alignItems="center">
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">default</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with placeholder</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with value</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with icon</Text>
      </XStack>
      
      <YStack space={10}>
        {/* Default row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">default</Text>
          <UrlInput aria-label='Default URL' />
          <UrlInput placeholder="Placeholder" aria-label='URL with placeholder' />
          <UrlInput value="https://rocket.chat" aria-label='URL with value' />
          <UrlInput 
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='URL with icon' 
          />
        </XStack>
        
        {/* Hover row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">hover</Text>
          <UrlInput aria-label='Hover URL' />
          <UrlInput placeholder="Placeholder" aria-label='Hover URL with placeholder' />
          <UrlInput value="https://rocket.chat" aria-label='Hover URL with value' />
          <UrlInput 
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='Hover URL with icon' 
          />
        </XStack>
        
        {/* Active row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">active</Text>
          <UrlInput aria-label='Active URL' />
          <UrlInput placeholder="Placeholder" aria-label='Active URL with placeholder' />
          <UrlInput value="https://rocket.chat" aria-label='Active URL with value' />
          <UrlInput 
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='Active URL with icon' 
          />
        </XStack>
        
        {/* Focus row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">focus</Text>
          <UrlInput aria-label='Focus URL' />
          <UrlInput placeholder="Placeholder" aria-label='Focus URL with placeholder' />
          <UrlInput value="https://rocket.chat" aria-label='Focus URL with value' />
          <UrlInput 
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='Focus URL with icon' 
          />
        </XStack>
        
        {/* Disabled row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">disabled</Text>
          <UrlInput disabled aria-label='Disabled URL' />
          <UrlInput disabled placeholder="Placeholder" aria-label='Disabled URL with placeholder' />
          <UrlInput disabled value="https://rocket.chat" aria-label='Disabled URL with value' />
          <UrlInput 
            disabled
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='Disabled URL with icon' 
          />
        </XStack>
        
        {/* Error row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">error</Text>
          <UrlInput error="Error" aria-label='Error URL' />
          <UrlInput error="Error" placeholder="Placeholder" aria-label='Error URL with placeholder' />
          <UrlInput error="Error" value="https://rocket.chat" aria-label='Error URL with value' />
          <UrlInput 
            error="Error"
            addon={<Icon name='discover' size={20} />} 
            value="https://rocket.chat" 
            aria-label='Error URL with icon' 
          />
        </XStack>
      </YStack>
    </YStack>
  ),
};

export const SmallStates: Story = {
  render: () => (
    <YStack space={10} alignItems="center">
      <XStack space={10} alignItems="center">
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">small</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with placeholder</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with value</Text>
        <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="center">with icon</Text>
      </XStack>
      
      <YStack space={10}>
        {/* Default row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">small</Text>
          <UrlInput small aria-label='Small URL' />
          <UrlInput small placeholder="Placeholder" aria-label='Small URL with placeholder' />
          <UrlInput small value="https://rocket.chat" aria-label='Small URL with value' />
          <UrlInput 
            small
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small URL with icon' 
          />
        </XStack>
        
        {/* Hover row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">hover</Text>
          <UrlInput small aria-label='Small hover URL' />
          <UrlInput small placeholder="Placeholder" aria-label='Small hover URL with placeholder' />
          <UrlInput small value="https://rocket.chat" aria-label='Small hover URL with value' />
          <UrlInput 
            small
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small hover URL with icon' 
          />
        </XStack>
        
        {/* Active row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">active</Text>
          <UrlInput small aria-label='Small active URL' />
          <UrlInput small placeholder="Placeholder" aria-label='Small active URL with placeholder' />
          <UrlInput small value="https://rocket.chat" aria-label='Small active URL with value' />
          <UrlInput 
            small
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small active URL with icon' 
          />
        </XStack>
        
        {/* Focus row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">focus</Text>
          <UrlInput small aria-label='Small focus URL' />
          <UrlInput small placeholder="Placeholder" aria-label='Small focus URL with placeholder' />
          <UrlInput small value="https://rocket.chat" aria-label='Small focus URL with value' />
          <UrlInput 
            small
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small focus URL with icon' 
          />
        </XStack>
        
        {/* Disabled row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">disabled</Text>
          <UrlInput small disabled aria-label='Small disabled URL' />
          <UrlInput small disabled placeholder="Placeholder" aria-label='Small disabled URL with placeholder' />
          <UrlInput small disabled value="https://rocket.chat" aria-label='Small disabled URL with value' />
          <UrlInput 
            small
            disabled
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small disabled URL with icon' 
          />
        </XStack>
        
        {/* Error row */}
        <XStack space={10} alignItems="center">
          <Text width={120} fontSize="$2" color="#9EA2A8" textAlign="left">error</Text>
          <UrlInput small error="Error" aria-label='Small error URL' />
          <UrlInput small error="Error" placeholder="Placeholder" aria-label='Small error URL with placeholder' />
          <UrlInput small error="Error" value="https://rocket.chat" aria-label='Small error URL with value' />
          <UrlInput 
            small
            error="Error"
            addon={<Icon name='discover' size={16} />} 
            value="https://rocket.chat" 
            aria-label='Small error URL with icon' 
          />
        </XStack>
      </YStack>
    </YStack>
  ),
};
