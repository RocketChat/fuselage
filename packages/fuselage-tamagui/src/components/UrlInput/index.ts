export { UrlInput } from './UrlInput';
export type { UrlInputProps } from './UrlInput';
