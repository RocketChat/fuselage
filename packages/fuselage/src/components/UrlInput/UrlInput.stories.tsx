import React from 'react';

import { Icon, UrlInput } from '../..';
import PropsVariation from '../../.storybook/PropsVariation';

export default {
  title: 'Forms/Inputs/UrlInput',
  component: UrlInput,
  parameters: {
    jest: ['UrlInput.spec.tsx'],
  },
};

export const Default = () => <UrlInput />;

export const WithIconAddon = () => (
  <UrlInput addon={<Icon name='send' size='x20' />} />
);

export const Invalid = () => <UrlInput error='Error' />;

export const Disabled = () => <UrlInput disabled />;

export const WithPlaceholder = () => <UrlInput placeholder='Placeholder' />;

export const WithValue = () => <UrlInput defaultValue='https://rocket.chat' />;

export const States = () => (
  <PropsVariation
    component={UrlInput}
    common={{ onChange: () => undefined }}
    xAxis={{
      'default': {},
      'with placeholder': { placeholder: 'Placeholder' },
      'with value': { value: 'https://rocket.chat' },
      'with icon': {
        addon: <Icon name='discover' size='x20' />,
        value: 'https://rocket.chat',
      },
    }}
    yAxis={{
      'default': {},
      'hover': { className: 'hover' },
      'active': { className: 'active' },
      'focus': { className: 'focus' },
      'disabled': { disabled: true },
      'errored': { error: 'Error' },
      'errored + hover': { className: 'hover', error: 'Error' },
      'errored + active': { className: 'active', error: 'Error' },
      'errored + focus': { className: 'focus', error: 'Error' },
    }}
  />
);
