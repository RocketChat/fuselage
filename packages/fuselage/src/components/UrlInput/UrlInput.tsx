import React, { ComponentProps, forwardRef, Ref } from 'react';

import { InputBox } from '../InputBox';

type UrlInputProps = Omit<ComponentProps<typeof InputBox>, 'type'>;

const UrlInput = forwardRef(function UrlInput(
  props: UrlInputProps,
  ref: Ref<HTMLElement>
) {
  return <InputBox type='url' ref={ref} {...props} />;
});

export default UrlInput;
