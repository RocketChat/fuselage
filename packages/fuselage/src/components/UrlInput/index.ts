export { default, default as UrlInput, UrlInputProps } from './UrlInput';
