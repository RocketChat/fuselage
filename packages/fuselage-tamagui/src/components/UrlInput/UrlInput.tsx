import type { ComponentProps, ReactNode, Ref } from 'react';
import { forwardRef } from 'react';

import { InputBox } from '../InputBox';

type UrlInputProps = Omit<ComponentProps<typeof InputBox>, 'type'> & {
  addon?: ReactNode;
  error?: string;
};

/**
 * An input for URL addresses.
 */
export const UrlInput = forwardRef(function UrlInput(
  props: UrlInputProps,
  ref: Ref<HTMLInputElement>,
) {
  return <InputBox type='url' ref={ref} {...props} />;
});
