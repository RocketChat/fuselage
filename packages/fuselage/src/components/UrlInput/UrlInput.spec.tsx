import { render } from '@testing-library/react';
import React from 'react';

import UrlInput from '.';

it('renders without crashing', () => {
  render(<UrlInput />);
});
