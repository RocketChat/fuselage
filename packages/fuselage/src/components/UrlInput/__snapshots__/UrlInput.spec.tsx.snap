// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UrlInput Component] renders without crashing 1`] = `
<body>
  <div>
    <input
      aria-label="url"
      class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-url rcx-input-box"
      size="1"
      type="url"
    />
  </div>
</body>
`;
